File: src/generators/UniformGenerator.py
import random
import sys

def generate_samples(num_samples, output_file):
    """Generate random numbers in [0,1) and save to file"""
    with open(output_file, 'w') as f:
        for _ in range(num_samples):
            value = random.random()  # [0,1)
            f.write(f"{value}\n")
    print(f"Generated {num_samples} samples in {output_file}")

if __name__ == "__main__":
    if len(sys.argv) < 3:
        print("Usage: python python_generator_01.py <num_samples> <output_file>")
        sys.exit(1)
    
    num_samples = int(sys.argv[1])
    output_file = sys.argv[2]
    
    generate_samples(num_samples, output_file)

File: src/generators/UniformGenerator.java
import java.io.FileWriter;
import java.io.IOException;
import java.util.Random;

public class UniformGenerator {
    public static void main(String[] args) {
        if (args.length < 2) {
            System.out.println("Usage: java JavaGenerator <num_samples> <output_file>");
            System.exit(1);
        }
        
        int numSamples = Integer.parseInt(args[0]);
        String outputFile = args[1];
        
        generateSamples(numSamples, outputFile);
    }
    
    public static void generateSamples(int numSamples, String outputFile) {
        Random random = new Random();
        
        try (FileWriter writer = new FileWriter(outputFile)) {
            writer.write(numSamples + "\n");
            writer.write("0 1\n");
            for (int i = 0; i < numSamples; i++) {
                double value = random.nextDouble(); // [0,1)
                writer.write(value + "\n");
            }
            System.out.println("Generated " + numSamples + " samples in " + outputFile);
        } catch (IOException e) {
            System.err.println("Error writing to file: " + e.getMessage());
        }
    }
}
